pkg/event-consumers/kafka/utils_test.go

```diff
@@ -0,0 +1,518 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kafka
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestGetTLSConfiguration(t *testing.T) {
+	caFile := buildCacertTempFile(t)
+	defer os.Remove(caFile.Name())
+	certFile := buildCertTempFile(t)
+	defer os.Remove(certFile.Name())
+	keyFile := buildKeyTempFile(t)
+	defer os.Remove(keyFile.Name())
+	fakeFile := buildFakeTempFile(t)
+	defer os.Remove(fakeFile.Name())
+
+	var tlsTests = []struct {
+		input struct {
+			caFile   string
+			certFile string
+			keyFile  string
+			insecure string
+		}
+		expected string
+	}{
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: certFile.Name(),
+				keyFile:  keyFile.Name(),
+				insecure: "",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: "",
+				keyFile:  keyFile.Name(),
+				insecure: "",
+			},
+			expected: ErrorCrtFileMandatory.Error(),
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: certFile.Name(),
+				keyFile:  "",
+				insecure: "",
+			},
+			expected: ErrorKeyFileMandatory.Error(),
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: "",
+				keyFile:  "",
+				insecure: "true",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: "",
+				keyFile:  "",
+				insecure: "",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   "",
+				certFile: "",
+				keyFile:  "",
+				insecure: "false",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   caFile.Name(),
+				certFile: "",
+				keyFile:  "",
+				insecure: "",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   fakeFile.Name(),
+				certFile: "",
+				keyFile:  "",
+				insecure: "",
+			},
+			expected: "",
+		},
+		{
+			input: struct {
+				caFile   string
+				certFile string
+				keyFile  string
+				insecure string
+			}{
+				caFile:   fmt.Sprintf("%s/%s", os.TempDir(), randomString(64)),
```

Comment:
is the `randomString` really necessary? wouldn't you get the same output with any nonexistent path?